### PR TITLE
Move modal to component and ES6

### PR DESCRIPTION
**Pull request by apastuhov**

### CODE REVIEW

File: src/directives/modal.js [hunk 1]
```diff
@@ -31,91 +31,159 @@ goog.require('ngeo');
...
-            modal.find('.modal-content').append(clone);
-          }
-        });
+  /**
+   * @private
+   */
+  onShow() {
+    this.childScope = this.$scope.$new();
+    // * @param {!function(!angular.Scope=, !function(Element)=)=} transcludeFn
+    // function((angular.Scope|null)=, function(angular.Scope, Element): ?=, (Element|null)=, string=): ?
```
Comment (pfirpfel) [inline src/directives/modal.js]: `// function((angular.Scope|null)=, function(angular.Scope, Element): ?=, (Element|null)=, string=): ?`

I'm a bit confused by this annotation.

**Comment by pfirpfel** [comment]
There are a bunch of files that still reference **ngeo.modalDirective**:

./examples/modal.html
./examples/modal.js
./contribs/gmf/src/directives/filterselector.js
./contribs/gmf/src/directives/editfeature.js
./contribs/gmf/src/directives/authenticationdirective.js
./contribs/gmf/examples/share.js

**Comment by pfirpfel** [comment]
The example is broken:
./examples/modal.html
./examples/modal.js

It just doesn't work, there is no error. I have not look at the example too thoroughly.

File: src/directives/modal.js (hunk 1 above)
Comment (apastuhov) [reply src/directives/modal.js]: It was an old annotation, I just forgot to remove it. Sorry for that)

**Comment by apastuhov** [comment]
@pfirpfel you are right.. I also did not found high-level problem. I will continue debugging on monday

File: src/directives/modal.js [hunk 2]
```diff
@@ -31,91 +31,161 @@ goog.require('ngeo');
...
-    template: '<div class="modal fade" tabindex="-1" role="dialog">' +
-        '<div class="modal-dialog">' +
-        '<div class="modal-content">' +
-        '</div>' +
-        '</div>' +
-        '</div>',
-    restrict: 'E',
-    require: 'ngModel',
-    transclude: true,
+ngeo.modalDirective;
```
Comment (fredj) [inline src/directives/modal.js]: namespace creation not needed

Comment (apastuhow) [reply src/directives/modal.js]: I keep it to have a doc on top

**Pull request merged**